python_ta/transforms/type_inference_visitor.py

```diff
@@ -334,8 +334,19 @@ def visit_index(self, node: astroid.Index) -> None:
         node.inf_type = node.value.inf_type
 
     def visit_slice(self, node: astroid.Slice) -> None:
-        # TODO: check input types by doing a typecheck for the slice constructor
-        node.inf_type = TypeInfo(slice)
+        lower_type = node.lower.inf_type.getValue() if node.lower else type(None)
+        upper_type = node.upper.inf_type.getValue() if node.upper else type(None)
+        step_type = node.step.inf_type.getValue() if node.step else type(None)
+        node.inf_type = self._handle_call(node, '__init__', slice, lower_type,
+                                          upper_type, step_type)
+        if node.inf_type.getValue() == type(None):
+            node.inf_type = TypeInfo(slice)
+
+    def visit_ext_slice(self, node: astroid.ExtSlice):
+        for dim in node.dims:
```

Comment:
This check isn't necessary; it's actually checking the structure of the AST, which is beyond the scope of our checker (which is all about types).

The `ExtSlice` node can be set to `NoType`; the more interesting algorithm part is in the `Subscript` node type-checking, where the "slice" attribute is an `ExtSlice`. That should be handled in a separate pull request, though.